src/pages/Filaments.tsx

```diff
@@ -76,7 +80,10 @@ const Filaments = () => {
                     <div className="flex items-center gap-x-6">
                       <button
                         className="text-gray-500 transition-colors duration-200 hover:text-red-500 focus:outline-none dark:text-gray-300 dark:hover:text-red-500"
-                        onClick={() => remove(f)}
+                        onClick={() => {
+                          setIsDialogOpen((pre) => !pre);
```

Comment:
No problem with your code; it is correct. The suggestion here is an improvement in code readability. What is expected when clicking on the icon? Opening the modal. Unless something is wrong in the modal implementation, there won't be a case where we end up closing the modal when clicking the delete button. Simply passing `true` will communicate a better intention for the future dev reading it.

```suggestion
                          setIsDialogOpen(true);
```

Reply:
All done @apbetioli